OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseUnitWebViewFragment.java

```diff
@@ -48,85 +40,11 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
-    }
-
-    @Override
-    public void onActivityCreated(Bundle savedInstanceState) {
-        super.onActivityCreated(savedInstanceState);
-
         authWebView.initWebView(getActivity(), true, false);
-        authWebView.getWebViewClient().setPageStatusListener(new URLInterceptorWebViewClient.IPageStatusListener() {
-            @Override
-            public void onPageStarted() {
-            }
-
-            @Override
-            public void onPageFinished() {
-                ViewPagerDownloadManager.instance.done(CourseUnitWebViewFragment.this, true);
-            }
-
-            @Override
-            public void onPageLoadError(WebView view, int errorCode, String description, String failingUrl) {
-                if (failingUrl != null && failingUrl.equals(view.getUrl())) {
-                    ViewPagerDownloadManager.instance.done(CourseUnitWebViewFragment.this, false);
-                }
-            }
-
-            @Override
-            @TargetApi(Build.VERSION_CODES.M)
-            public void onPageLoadError(WebView view, WebResourceRequest request, WebResourceResponse errorResponse,
-                                        boolean isMainRequestFailure) {
-                if (isMainRequestFailure) {
-                    ViewPagerDownloadManager.instance.done(CourseUnitWebViewFragment.this, false);
-                }
-            }
-
-            @Override
-            public void onPageLoadProgressChanged(WebView view, int progress) {
-            }
-        });
-
-        if (ViewPagerDownloadManager.USING_UI_PRELOADING) {
-            if (ViewPagerDownloadManager.instance.inInitialPhase(unit)) {
-                ViewPagerDownloadManager.instance.addTask(this);
-            } else {
-                authWebView.loadUrl(true, unit.getBlockUrl());
-            }
-        }
-    }
-
-    @Override
-    public void run() {
-        if (this.isRemoving() || this.isDetached()) {
-            ViewPagerDownloadManager.instance.done(this, false);
-        } else {
-            authWebView.loadUrl(true, unit.getBlockUrl());
-        }
+        authWebView.loadUrl(true, unit.getBlockUrl());
     }
 
-    //the problem with viewpager is that it loads this fragment
-    //and calls onResume even it is not visible.
-    //which breaks the normal behavior of activity/fragment
-    //lifecycle.
-    @Override
-    public void setUserVisibleHint(boolean isVisibleToUser) {
-        super.setUserVisibleHint(isVisibleToUser);
-        if (ViewPagerDownloadManager.USING_UI_PRELOADING) {
-            return;
-        }
-        if (ViewPagerDownloadManager.instance.inInitialPhase(unit)) {
-            return;
-        }
-        if (isVisibleToUser) {
-            if (authWebView != null) {
-                authWebView.loadUrl(false, unit.getBlockUrl());
-            }
-        } else if (authWebView != null) {
-            authWebView.tryToClearWebView();
-        }
-    }
-
-    @Override
+    /*@Override
     public void onResume() {
```

Comment:
`onResume` method needs to uncomment and change as

`    
@Override
    public void onResume() {
        super.onResume();
        authWebView.onResume();
    }
`

Reply:
good catch!